### PR TITLE
Parsing  the class location iteratively 

### LINKED ISSUE #211: Make references Dynamic
### I'm submitting a 
- [X] bug report.
- [ ] feature request.

### Current Behaviour:

Right now we are only able to accept semantic refs which are only 1 level deep , like '#definitions/className" however we should have done this dynamically so that we can accept refs like "#definitions/location1/location1.location2/location1.location2.location3/className" and so on .
### Expected Behaviour:

We would be able to parse refs of dynamic length .
### Do you want to work on this issue?

yes , working on it !

### PR DESCRIPTION
**Pull request by vaibhavchellani**
Fixes #211

### Checklist
- [X] My branch is up-to-date with upstream/develop branch.
- [X] Everything works and tested for Python 3.5.2 and above.

### Description

The parser can now go to any level deep to parse class or other objects .Earlier we were only able to go 1 level deep now we are parsing iteratively to fetch data.
### Test Logs
screen shot 2018-06-18 at 1 28 30 pm

### CODE REVIEW

File: hydrus/parser/openapi_parser.py [hunk 1]
```diff
@@ -22,6 +22,19 @@ def try_catch_replacement(block: Any, get_this: str, default: Any) -> str:
         return default
 
 
+def get_class_name(class_location):
+    return class_location[len(class_location)-1]
+
+
+def get_data_at_location(class_location, doc):
```
Comment (xadahiya) [inline hydrus/parser/openapi_parser.py]: Use mypy format and add proper DocStrings

Comment (vaibhavchellani) [reply hydrus/parser/openapi_parser.py]: @xadahiya fixed , other mypy errors have been fixed in #210

File: hydrus/parser/openapi_parser.py [hunk 2]
```diff
@@ -22,6 +22,30 @@ def try_catch_replacement(block: Any, get_this: str, default: Any) -> str:
...
+
+    :param class_location: list containing the class location
+    :return: name of class
+    """
+    return class_location[len(class_location)-1]
+
+
+def get_data_at_location(class_location: List[str], doc: Dict[str,Any]) -> Dict[str,Any]:
+    """
+
```
Comment (xadahiya) [inline hydrus/parser/openapi_parser.py]: Add proper Docstrings explaining what each function does.

File: hydrus/parser/openapi_parser.py [hunk 3]
```diff
@@ -224,11 +284,17 @@ def get_paths(doc: Dict["str", Any], classAndClassDefinition: Dict["str",HydraCl
...
-                            api_doc.add_supported_class(classAndClassDefinition[class_name], collection=False)
+                            api_doc.add_supported_class(
+                                classAndClassDefinition[class_name], collection=False)
     generateEntrypoint(api_doc)
 
 
-def parse(doc):
+def parse(doc: Dict[str, Any]) -> str:
+    """
+    parent function for parsing the open api documentation
```
Comment (xadahiya) [inline hydrus/parser/openapi_parser.py]: @vaibhavchellani  You should also use proper punctuations for function explanations in Docstrings.

**Comment by vaibhavchellani** [comment]
@xadahiya please let me know if there are any other changes or if the quality standards are not being met .

**Pull request merged**